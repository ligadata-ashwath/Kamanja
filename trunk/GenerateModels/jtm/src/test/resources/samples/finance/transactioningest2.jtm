{
  "header": {
    "namespace": "com.ligadata.models.samples.finance",
    "version": "0.0.1",
    "language" : "scala",
    "minVersion" : "2.11",
    "description": "Finance load transaction message",
    "name": "TransactionIngest"
  },
  "imports": {
    "packages": [ "com.ligadata.runtime.Conversion"]
  },
  "aliases": {
    "messages": {
      "m1": "com.ligadata.kamanja.samples.messages.TransactionMsgIn",
      "o1": "com.ligadata.kamanja.samples.messages.TransactionMsg"
    }
  },
  "transformations": {
    "transactionmsg": {
      "dependsOn": [ ["m1"] ],
      "computes": {
        "mapData":{ "type": "System.MapOfString", "val": "KeyValueMap($data, \"|\", \"=\")", "comment": "Split the incoming data" }
      },
      "outputs": {
        "o1": {
          "computes": {
            "typeName": { "type": "String", "val": "$mapData.typename", "comment": "extract the type" }
          },
          "where": "\"com.ligadata.kamanja.samples.messages.TransactionMsg\" == $typeName",
          "mapping": {
            "custid": "$mapData.custid",
            "branchid": "$mapData.branchid",
            "accno": "$mapData.accno",
            "amount": "$mapData.amount",
            "balance": "$mapData.balance",
            "date": "$mapData.date",
            "time": "$mapData.time",
            "locationid": "$mapData.locationid",
            "transtype": "$mapData.transtype"
          }
        }
      }
    }
  }
}
