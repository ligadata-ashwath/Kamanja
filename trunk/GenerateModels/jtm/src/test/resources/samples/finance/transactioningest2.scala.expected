/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.models.samples.finance.V1
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.runtime.Log
import com.ligadata.runtime.Conversion
class TransactionIngestFactory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  override def createModelInstance(): ModelInstance = return new TransactionIngest(this)
  override def getModelName: String = "com.ligadata.models.samples.finance.TransactionIngest"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
}
class TransactionIngest(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  val conversion = new com.ligadata.runtime.Conversion
  val log = new com.ligadata.runtime.Log(this.getClass.getName)
  val context = new com.ligadata.runtime.JtmContext
  import log._
  // Model code start
  // Model code end
  override def execute(txnCtxt: TransactionContext, execMsgsSet: Array[ContainerOrConcept], triggerdSetIndex: Int, outputDefault: Boolean): Array[ContainerOrConcept] = {
    if (isTraceEnabled)
    Trace(s"Model::execute transid=%d triggeredset=%d outputdefault=%s".format(txnCtxt.transId, triggerdSetIndex, outputDefault.toString))
    if(isDebugEnabled)
    {
      execMsgsSet.foreach(m => Debug( s"Input: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    // Grok parts
    // Model methods
    def exeGenerated_transactionmsg_1(msg1: com.ligadata.kamanja.samples.messages.V1000000.TransactionMsgIn): Array[MessageInterface] = {
      Debug("exeGenerated_transactionmsg_1")
      // Split the incoming data
      val mapData: System.MapOfString = KeyValueMap(msg1.data, "|", "=")
      def process_o1(): Array[MessageInterface] = {
        Debug("exeGenerated_transactionmsg_1::process_o1")
        // extract the type
        val typeName: String = mapData.typename
        if (!("com.ligadata.kamanja.samples.messages.TransactionMsg" == typeName)) {
          Debug("Filtered: transactionmsg@o1")
          return Array.empty[MessageInterface]
        }
        val result = com.ligadata.kamanja.samples.messages.V1000000.TransactionMsg.createInstance
        result.branchid = conversion.ToInteger(mapData.get("branchid"))
        result.custid = conversion.ToLong(mapData.get("custid"))
        result.locationid = conversion.ToInteger(mapData.get("locationid"))
        result.balance = conversion.ToDouble(mapData.get("balance"))
        result.amount = conversion.ToDouble(mapData.get("amount"))
        result.transtype = mapData.get("transtype")
        result.accno = conversion.ToLong(mapData.get("accno"))
        result.date = conversion.ToInteger(mapData.get("date"))
        result.time = conversion.ToInteger(mapData.get("time"))
        Array(result)
      }
      process_o1()
    }
    // Evaluate messages
    val msgs = execMsgsSet.map(m => m.getFullTypeName -> m).toMap
    val msg1 = msgs.getOrElse("com.ligadata.kamanja.samples.messages.TransactionMsgIn", null).asInstanceOf[com.ligadata.kamanja.samples.messages.V1000000.TransactionMsgIn]
    // Main dependency -> execution check
    // Create result object
    val results: Array[MessageInterface] =
    (if(msg1!=null) {
      exeGenerated_transactionmsg_1(msg1)
    } else {
      Array.empty[MessageInterface]
    }) ++
    Array.empty[MessageInterface]
    if(isDebugEnabled)
    {
      results.foreach(m => Debug( s"Output: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    results.asInstanceOf[Array[ContainerOrConcept]]
  }
}
